tests/workspace_svg/workspace_svg_test.js

```diff
@@ -24,6 +24,12 @@ function helper_createWorkspaceWithToolbox() {
   return Blockly.inject('blocklyDiv', {toolbox: toolbox});
 }
 
+function helper_createNewBlock(workspace, type) {
+  var block = workspace.newBlock(type);
+  block.initSvg();
```

Comment:
Where does this get called outside of testing? Should WorkspaceSvg.newBlock() override Workspace.newBlock() to provide this functionality?

Reply:
It gets called all over the code base. It might make sense to do a cleanup and move it into BlockSvg's constructor, but I'm not sure if there would be side effects.

Reply:
I've added #1980 to to track this issue.

Reply:
Alllll over the place.  I've chatted with Neil about this and we agree that it's worth cleaning up eventually, but carefully.

Recall that Blockly did not originally have a headless mode.  When creating the headless mode, Neil pulled out some common code and had one XML helper function call the other.  The rendered version then goes through after the fact and does some additional work to put everything in order.  The main problem is that deserializing a block may actually create multiple blocks down the stack, and the information flow doesn't really reflect that.  

The solution is probably to completely separate the code for deserializing rendered and non-rendered blocks, which will then let us do less post-facto messing around to get SVG initialization right.  The comment serialization/deserialization stuff has a similar issue.